src/utils/utils_level_zero.h

```diff
@@ -9,6 +9,7 @@
 #define UMF_UTILS_LEVEL_ZERO_H
 
 #include <umf/providers/provider_level_zero.h>
+#include <vector>
```

Comment:
should this be moved below to the __cplusplusplus section?

Reply:
No, includes should not be under that ifdef

Reply:
I didn't notice this was not a c++-only header file. I changed the function to not use any c++ types and just return the number of properties.